src/main/java/com/example/NiRi/UserController.java

```diff
@@ -21,4 +24,24 @@ public User registerUser(@RequestBody User user) {
     public List<User> getAllUsers() {
         return userService.getAllUsers();
     }
-}
+
+    @GetMapping("/getByEmail")
+    public User getUserByEmail(@RequestParam String email) {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new RuntimeException("User not found with email: " + email));
+    }
+
+    @PutMapping("/updateUser/{id}")
+    public User updateUserDetails(@PathVariable Long id, @RequestBody User updatedUser) {
```

Comment:
Can you move the logic from this function to your UserService

Reply:
Done.